vpr/src/route/rr_graph_storage.cpp

```diff
@@ -492,20 +492,19 @@ void t_rr_graph_storage::partition_edges() {
     if (partitioned_) {
         return;
     }
-
+    const auto& device_ctx = g_vpr_ctx.device();
     edges_read_ = true;
     VTR_ASSERT(remapped_edges_);
-
-    const auto& device_ctx = g_vpr_ctx.device();
     // This sort ensures two things:
     //  - Edges are stored in ascending source node order.  This is required
     //    by assign_first_edges()
     //  - Edges within a source node have the configurable edges before the
     //    non-configurable edges.
+    /*TODO: The API should be moved to ``RRGraphBuilder`` because it required rr_switch which is not included in node_storage */
     std::sort(
         edge_sort_iterator(this, 0),
         edge_sort_iterator(this, edge_src_node_.size()),
-        edge_compare_src_node_and_configurable_first(device_ctx.rr_switch_inf));
+        edge_compare_src_node_and_configurable_first(g_vpr_ctx.mutable_device().rr_graph_builder.rr_switch_inf()));
```

Comment:
Double check if you do need a mutable rr_switch vector here. If not, just use RRGraphView. It is safe.

Reply:
We Can not access  `rr_graph_view` here. You suggested me to add comments here we will move this to `rr_graph_builder`.

Reply:
My bad. No problem then.